synapse/models/math.py

```diff
@@ -0,0 +1,50 @@
+import synapse.lib.module as s_module
+
+class MathModule(s_module.CoreModule):
+
+    def getModelDefs(self):
+        return (('math', {
+            'types': (
+
+                ('math:algorithm', ('guid', {}), {
+                    'doc': 'A mathematical algorithm.'}),
+
+                ('math:algorithm:type:taxonomy', ('taxonomy', {}), {
+                    'interfaces': ('meta:taxonomy',),
+                    'doc': 'A hierarchical taxonomy of algorithm types.'}),
+            ),
+            'edges': (
+
+                (('risk:tool:software', 'uses', 'math:algorithm'), {
+                    'doc': 'The tool uses the algorithm.'}),
+
+                (('it:prod:softver', 'uses', 'math:algorithm'), {
+                    'doc': 'The software uses the algorithm.'}),
+
+                (('file:bytes', 'uses', 'math:algorithm'), {
+                    'doc': 'The file uses the algorithm.'}),
+
+                (('math:algorithm', 'generated', None), {
+                    'doc': 'The target node was generated by the algorithm.'}),
+            ),
+            'forms': (
+
+                ('math:algorithm:type:taxonomy', {}, ()),
+
+                ('math:algorithm', {}, (
```

Comment:
If it's "just" an algo name ("MD5", "Hamming distance") does this need to be a guid vs str (like crypto:algorithm)? Guid is more flexible but is it necessary?

Reply:
yep! it's necessary because we will likely end up with an alternate names array and entity resolution requirements. Simple primary string properties always seem to get us into trouble :)